presto-common/src/main/java/com/facebook/presto/common/zorder/ZOrderIntegers.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.zorder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class ZOrderIntegers
+{
+    private ZOrderIntegers() {}
+
+    // TODO: make int and long z-address version
+    public static byte[] encode(List<Integer> input, List<Integer> bitRanges)
```

Comment:
bitRanges could be confusing since range normally means min and max. leftmostSetBitPositions sounds a little bit better. You can think of a better name.